### PR TITLE
SCA implementation

### PR DESCRIPTION
**Pull request by valentinDamyanovAmpeco**
upd: purchase request with automatic confirmation so we can use confirmPayment in the stripe-react-native, add helper response methods to detect if authorization required.

upd: handle sea rest response.

add: handle payment_intent.payment_failed IPN 

fix: remove unnecessary code line

### CODE REVIEW

File: src/Message/NotificationResponse.php [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?php
+
+namespace Omnipay\Stripe\Message;
+
+class NotificationResponse extends Response
```
Comment (dmtar) [inline src/Message/NotificationResponse.php]: I'm not sure we are using this anywhere, but it seems more correct naming than RestResponse if it's just used for notification response

File: src/Message/PurchaseRequest.php [hunk 2]
```diff
@@ -70,6 +70,7 @@ public function getData()
     {
         $data = parent::getData();
         $data['capture'] = 'true';
+        $data['confirmation_method'] = 'automatic';
```
Comment (dmtar) [inline src/Message/PurchaseRequest.php]: confirmation_method is relevant only for payment intents, the old charges API is not used at all. We only use the new payment intents API - PaymentIntentsGateway

File: src/Message/RestResponse.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Omnipay\Stripe\Message;
+
+use Omnipay\Common\Message\AbstractResponse;
+use Omnipay\Common\Message\RequestInterface;
+
+class RestResponse extends AbstractResponse
```
Comment (dmtar) [inline src/Message/RestResponse.php]: Directly extend the Response and override only the isSuccessful, getTransactionReference is already implemented there

**Review by dmtar — CHANGES_REQUESTED** [review]

File: src/Message/RestResponse.php (hunk 3 above)
Comment (valentinDamyanovAmpeco) [reply src/Message/RestResponse.php]: our method `getTransactionReference` handles  `$this->data['data']['object']['charges']['data'][0]['id'] ?? null;`
while inherited have different usage

**Pull request merged**